corehq/apps/api/odata/tests/test_feed.py

```diff
@@ -220,3 +224,107 @@ def _get_correct_credentials(cls):
 @flag_enabled('TWO_FACTOR_SUPERUSER_ROLLOUT')
 class TestFormOdataFeedWithTwoFactorUsingApiKey(TestFormOdataFeedUsingApiKey):
     pass
+
+
+class TestCaseOdataFeedFromExportInstance(TestCase, OdataTestMixin):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestCaseOdataFeedFromExportInstance, cls).setUpClass()
+        cls._set_up_class()
+        cls._setup_accounting()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls._teardownclass()
+        cls._teardown_accounting()
+        super(TestCaseOdataFeedFromExportInstance, cls).tearDownClass()
+
+    def test_no_credentials(self):
+        with flag_enabled('ODATA'):
+            response = self.client.get(self.view_url)
+        self.assertEqual(response.status_code, 401)
+
+    def test_wrong_password(self):
+        wrong_credentials = self._get_basic_credentials(self.web_user.username, 'wrong_password')
+        with flag_enabled('ODATA'):
+            response = self._execute_query(wrong_credentials)
+        self.assertEqual(response.status_code, 401)
+
+    def test_wrong_domain(self):
+        other_domain = Domain(name='other_domain')
+        other_domain.save()
+        self.addCleanup(other_domain.delete)
+
+        correct_credentials = self._get_correct_credentials()
+        with flag_enabled('ODATA'):
+            response = self.client.get(
+                self._odata_feed_url_by_domain(other_domain.name),
+                HTTP_AUTHORIZATION='Basic ' + correct_credentials,
+            )
+        self.assertEqual(response.status_code, 401)
+
+    def test_missing_feature_flag(self):
+        correct_credentials = self._get_correct_credentials()
+        response = self._execute_query(correct_credentials)
+        self.assertEqual(response.status_code, 404)
```

Comment:
It seems like every test case you write has the same 5-6 error cases. Just confirming you've thought about whether they can be genericized so that there's less code and decided it's not worth it?

Reply:
There was some talk about that here:
https://github.com/dimagi/commcare-hq/pull/24110#discussion_r277804948

The existing approach was chosen to avoid a complex inheritance pattern and to handle small differences between the tests.